Refactor the Popover component to use the floatingUI library (#40740)
WordPress/gutenberg: packages/components/src/border-box-control/border-box-control/README.md

- Required: Yes

> ### `popoverPlacement`: `Object`

Reviewer: ```suggestion
### `popoverPlacement`: `string`
```

Looks like the type for `popoverPlacement` is now `string`.

The position of the color popover relative to the control wrapper.

Example:
```js
{
	popoverPlacement: "left-start"

> }

Reviewer: Looks like this example should be removed or replaced given the prop's type is actually a string and not an object.

Perhaps we could detail the valid placement/position values available?

```

By default, popovers are displayed relative to the button that initiated the popover. By supplying a popover placement, you force the popover to display in a specific location.

- Required: No

> ### `popoverOffset`: `Object`

Reviewer: ```suggestion
### `popoverOffset`: `number`
```

This should match the defined type.

> Works in conjunctions with `popoverPlacement` and allows to leave a space between the color popover and the control wrapper.

Reviewer: ```suggestion
Works in conjunctions with `popoverPlacement` and allows leaving a space between the color popover and the control wrapper.
```

nit: Above suggestion is just slight smoother wording in my opinion 🤷

Example:
```js
{
	popoverOffset: 10
}

> ```

Reviewer: Is this example still warranted when the prop is a simple number?
